src/content/docs/en/guides/backend/supabase.mdx

```diff
@@ -3,10 +3,806 @@ title: Supabase & Astro
+```
+
+To add Supabase credentials to Astro, create an `.env` file in the root of the project with the following:
+
+```ini
+# .env
+SUPABASE_URL=SUPABASE_URL
+SUPABASE_ANON_KEY=SUPABASE_ANON_KEY
+SUPABASE_JWT_SECRET=SUPABASE_JWT_SECRET
+```
+
+The Project structure should now include this file:
+
+<FileTree title="Project Structure">
+- ...
+- supabase/
+- .env
+- ...
+</FileTree>
+
+Now, these environment variables are available for use in the project.
+
+To have IntelliSense for the Supabase environment variables, edit or create the file `env.d.ts` in `src/` directory and configure the types:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly SUPABASE_URL: string;
+  readonly SUPABASE_ANON_KEY: string;
+  readonly SUPABASE_JWT_SECRET: string;
+}
+
+interface ImportMeta {
+  readonly env: ImportMetaEnv;
+}
+```
+
+:::tip
+Read more about [environment variables](/en/guides/environment-variables/) and `.env` files in Astro.
+:::
+
+The project structure should include these files:
+<FileTree title="Project Structure">
+- ...
+- supabase/
+- src/
+  - env.d.ts
+- .env
+- ...
+</FileTree>
+
+### Install Dependencies
+
+To connect Astro with Supabase, install the following package using the command below:
+
+- `@supabase/supabase-js`
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+    ```shell
+    npm install @supabase/supabase-js
+    ```
+  </Fragment>
+  <Fragment slot="pnpm">
+    ```shell
+    pnpm install @supabase/supabase-js
+    ```
+  </Fragment>
+  <Fragment slot="yarn">
+    ```shell
+    yarn add @supabase/supabase-js
+    ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Create Profile Table
+
+1. Go to [Studio URL](http://localhost:54323)
+2. Click on SQL Editor on the left sidebar
+3. Choose `User Management Starter` from `Quick start` in the bottom of the page
+4. Click `RUN`
+
+### Create Supabase Client
+
+1. Create a directory named `lib` in the `src/` directory.
+2. Create a file named `supabaseClient.ts` in the `src/lib/` directory.
+
+In `supabaseClient.ts` add the following code to create the supabase client.
+
+```ts title="src/lib/supabaseClient.ts"
+import { createClient } from '@supabase/supabase-js'
+
+export const supabase = createClient(
+  `${import.meta.env.SUPABASE_URL}`,
+  `${import.meta.env.SUPABASE_ANON_KEY}`,
+  {         
+    auth: {
+      persistSession: false,
+    }
+  }
+)
+```
+
+The project structure should include these new files:
+
+<FileTree title="Project Structure">
+- ...
+- src/
+  - lib/
+    - supabaseClient.ts
+  - env.d.ts
+- .env
+- astro.config.mjs
+- package.json
+- ...
+</FileTree>
+
+### Prepare Authentication Functions
+
+- Create a file named `auth.ts` in the `src/lib/` directory.
+
+In `auth.ts` add the following code to create the authentication functions:
+
+```ts title="src/lib/auth.ts"
```

Comment:
As a good practice, we should try to minimize the use utility functions. Even though we will repeat a lot of code, it helps users see which functions we are using from the original library.